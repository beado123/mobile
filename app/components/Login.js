import React, {Component} from 'react';
import { StyleSheet, Text, View, ImageBackground, Button, TouchableOpacity, TextInput, Linking, AsyncStorage } from 'react-native';
import { List, ListItem } from "react-native-elements";
import { Constants } from 'expo';
import styleLogin from '/home/yihan/AwesomeProject/app/styles/Login.js';
import styleHeader from '/home/yihan/AwesomeProject/app/styles/Header.js';

{/*The design of login page follows from https://www.youtube.com/watch?v=jGU2I6sYDLE*/}
export default class FollowersPage extends React.Component {

    state = {
        username: '',
        psw: ''
    };

    /*This function stores username and password to AsyncStorage and navigates to profile page. */
    login = () => {
        AsyncStorage.setItem('username', this.state.username);
        AsyncStorage.setItem('password', this.state.psw);
        this.props.navigation.navigate('profile');
    };

    render() {
        return (
            <View style={styleLogin.container}>
                <ImageBackground style={styleHeader.headerBackground} source={require('/home/yihan/AwesomeProject/app/img/wallpaper2.jpg')}>
                    <View style={styleLogin.content}>
                        <Text style={styleLogin.logo}>Try your Github account!</Text>
                            <View style={styleLogin.inputContainer}>
                                <TextInput style={styleLogin.input} onChangeText={(text) => this.setState({username:text})} placeholder='username'></TextInput>
                                <TextInput secureTextEntry={true} style={styleLogin.input} onChangeText={(text) => this.setState({psw:text})} placeholder='password'></TextInput>
                                <TouchableOpacity style = {styleLogin.buttonLogin} onPress = {() => this.login()}>
                                    <Text style={styleLogin.buttonText}>LOGIN</Text>
                                </TouchableOpacity>
                            </View>
                    </View>
                </ImageBackground>
            </View>
        );
    }
}
