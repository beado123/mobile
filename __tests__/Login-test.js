import React from 'react';
import "isomorphic-fetch"
import { StackNavigator } from 'react-navigation';
import Login from '/home/yihan/AwesomeProject/app/components/Login.js';
import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer.create(<Login />).toJSON();
  expect(tree).toMatchSnapshot();
});
