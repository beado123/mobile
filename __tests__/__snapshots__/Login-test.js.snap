// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "flex": 1,
      }
    }
  >
    <Image
      source={1}
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": undefined,
            "width": undefined,
          },
          undefined,
        ]
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
            "fontFamily": "Georgia",
            "fontSize": 25,
            "marginTop": 80,
            "textAlign": "center",
            "textShadowColor": "#429DD4",
            "textShadowOffset": Object {
              "height": 2,
              "width": 2,
            },
            "textShadowRadius": 15,
            "width": 400,
          }
        }
      >
        Try your Github account!
      </Text>
      <View
        style={
          Object {
            "alignSelf": "stretch",
            "borderColor": "#ffff",
            "borderWidth": 1,
            "margin": 70,
            "marginBottom": 0,
            "padding": 20,
            "paddingBottom": 10,
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          placeholder="username"
          style={
            Object {
              "backgroundColor": "#DDEDF6",
              "fontFamily": "Palatino",
              "fontSize": 16,
              "height": 40,
              "marginBottom": 10,
              "padding": 10,
            }
          }
        />
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          placeholder="password"
          secureTextEntry={true}
          style={
            Object {
              "backgroundColor": "#DDEDF6",
              "fontFamily": "Palatino",
              "fontSize": 16,
              "height": 40,
              "marginBottom": 10,
              "padding": 10,
            }
          }
        />
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignSelf": "stretch",
              "backgroundColor": "#BFDDEF",
              "borderColor": "#fff",
              "borderWidth": 1,
              "margin": 10,
              "opacity": 1,
              "padding": 10,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontFamily": "Georgia",
                "fontSize": 16,
                "fontWeight": "bold",
                "textAlign": "center",
              }
            }
          >
            LOGIN
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
